be/src/exprs/vectorized/utility_functions.cpp

```diff
@@ -63,36 +68,55 @@ ColumnPtr UtilityFunctions::last_query_id(FunctionContext* context, const Column
     }
 }
 
-ColumnPtr UtilityFunctions::uuid(FunctionContext*, const Columns& columns) {
+ColumnPtr UtilityFunctions::uuid(FunctionContext* ctx, const Columns& columns) {
     int32_t num_rows = ColumnHelper::get_const_value<TYPE_INT>(columns.back());
 
-    ColumnBuilder<TYPE_VARCHAR> result(num_rows);
+    auto col = UtilityFunctions::uuid_numeric(ctx, columns);
+    auto& uuid_data = down_cast<Int128Column*>(col.get())->get_data();
+
+    auto res = BinaryColumn::create();
+    auto& bytes = res->get_bytes();
+    auto& offsets = res->get_offset();
+
+    offsets.resize(num_rows + 1);
+    bytes.resize(33 * num_rows);
+
+    char* ptr = reinterpret_cast<char*>(bytes.data());
     for (int i = 0; i < num_rows; ++i) {
-        result.append(generate_uuid_string());
```

Comment:
I think respectively handle offsets array and uuid_data array will get better performance